localstack/testing/pytest/fixtures.py

```diff
@@ -1978,3 +1978,54 @@ def factory(**kwargs):
             appsync_client.delete_graphql_api(apiId=api)
         except Exception as e:
             LOG.debug(f"Error cleaning up AppSync API: {api}, {e}")
+
+
+@pytest.fixture
+def assert_host_customisation(monkeypatch):
+    hostname_external = f"external-host-{short_uid()}"
+    localstack_hostname = f"localstack-hostname={short_uid()}"
+    monkeypatch.setattr(config, "HOSTNAME_EXTERNAL", hostname_external)
+    # monkeypatch.setattr(config, "LOCALSTACK_HOSTNAME", localstack_hostname)
```

Comment:
I'm struggling with this one. If I uncomment this line, two tests fail:
* opensearch testing the path strategy, and
* lambda testing the legacy http api to generate the function url.

This is because
* opensearch starts the domain with the `LOCALSTACK_HOSTNAME` hostname as its url, and therefore the health check keeps failing since the nonsense  value generated here is not a valid hostname, so LS can't check the domain is up
* a similar story with the lambda endpoint - requests are made to the nonsense hostname as part of the test, which is not a valid hostname.

Though it's documented as read-only, we do allow the user to change `LOCALSTACK_HOSTNAME`, hence why I'm trying to test it here. Since it has to be a valid host (and `localhost` doesn't really count) I'm not sure how to allow this line to be uncommented. Any thoughts?

Reply:
One idea is to set the value to `socket.gethostname()` which should always return a host that's accessible, and is (probably) not `localhost`, which would probably fulfil this criterion.

Reply:
Using `socket.gethostname` works nicely